internal/app/backend/synchronizer_client.go

```diff
@@ -43,72 +38,19 @@ func newSynchronizerClient(cfg config.View) *synchronizerClient {
 	}
 }
 
-// register calls the Register method on Synchronizer service. It only triggers this
-// if the synchronizer is enabled. The first attempt to call the synchronizer service
-// establishes a connection. Consequent requests use the cached connection.
-func (sc *synchronizerClient) register(ctx context.Context) (string, error) {
-	if !sc.enabled() {
-		// Synchronizer is disabled. Succeed the call without returning any ID.
-		return "", nil
-	}
-
-	client, err := sc.cacher.Get()
-	if err != nil {
-		return "", err
-	}
-
-	resp, err := client.(ipb.SynchronizerClient).Register(ctx, &ipb.RegisterRequest{})
-	if err != nil {
-		return "", err
-	}
-
-	return resp.GetId(), nil
+type synchronizerStream interface {
+	Send(*ipb.SynchronizeRequest) error
+	Recv() (*ipb.SynchronizeResponse, error)
+	CloseSend() error
 }
 
-// evaluate calls the EvaluateProposals method on Synchronizer service. It only triggers
-// this if the synchronizer is enabled. The first attempt to call the synchronizer service
-// establishes a connection. Consequent requests use the cached connection.
-func (sc *synchronizerClient) evaluate(ctx context.Context, id string, proposals []*pb.Match) ([]*pb.Match, error) {
-	if !sc.enabled() {
-		// Synchronizer is disabled. Return all the proposals as results. This is only temporary.
-		// After the synchronizer is implememnted, it will be mandatory and this check will be removed.
-		return proposals, nil
-	}
-
-	client, err := sc.cacher.Get()
-	if err != nil {
-		return nil, err
-	}
-
-	ctx, err = util.AppendSynchronizerContextID(ctx, id)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "%v", err)
-	}
-
-	stream, err := client.(ipb.SynchronizerClient).EvaluateProposals(ctx)
-	for _, proposal := range proposals {
-		if err = stream.Send(&ipb.EvaluateProposalsRequest{Id: id, Match: proposal}); err != nil {
+func (sc *synchronizerClient) synchronize(ctx context.Context) (synchronizerStream, error) {
```

Comment:
It looks a little bit weird to me that we need to define a synchronizerStream interface to use the stream. Is it possible to implement a `func getClient() (ipb.SynchronizerClient, error)` function and use it directly?

Reply:
We need to have a discussion on whether disabling the synchronizer is still allowed.  I didn't return the stream directly (and in that case there's no reason to have this file really.) because that way we could provide a dummy synchronize service.  It would have to add tickets to the ignore list though.

Reply:
Per discussion, removed the config option altogether.